### ISSUE TITLE
Missing Command-line Information for Key Baseline Networks in Model Zoo

### ISSUE THREAD

**Issue by jwook1004** [NervanaSystems/distiller#170]
Hi,

Thanks for providing a great framework for DNN pruning and quantization. I really like a wide variety of algorithms implemented into the Model Zoo. 

One critical thing missed here is the command-line information for the baseline networks in the Model zoo. To reproduce the results for the models provided in "[distiller/examples](https://github.com/NervanaSystems/distiller/tree/master/examples)", having the scheduler files (i.e., YAML) is not enough; they do not have information about key hyper-parameters such as initial learning rate, weight decay factor, minibatch size and the total number of epochs. Especially, such info seems to be missed for imagenet tasks, especially the most critical baseline networks: [alexnet](https://github.com/NervanaSystems/distiller/blob/master/examples/quantization/fp32_baselines/alexnet_bn_base_fp32.yaml) and [resnet18](https://github.com/NervanaSystems/distiller/blob/master/examples/quantization/fp32_baselines/preact_resnet18_imagenet_base_fp32.yaml). 

Since many pruning works implemented here use alexnet and resnet18 for Imagenet demo, it would be critical to know the baseline command-line info for reproduction of the proper baseline. (It would save significant time and effort to find the hyper-params and expedite adaptation of Distiller for various research.) Can anyone share some help on this?

Best regards,
Jungwook

**Comment by nzmora** [comment]
Hi Jungwook,

You're right - some of the example YAML files don't have all the information (some do - e.g. [this](https://github.com/NervanaSystems/distiller/blob/master/examples/agp-pruning/mobilenet_imagenet_baseline_training.yaml)).

Regarding [alexnet](https://github.com/NervanaSystems/distiller/blob/master/examples/quantization/fp32_baselines/alexnet_bn_base_fp32.yaml) and [resnet18](https://github.com/NervanaSystems/distiller/blob/master/examples/quantization/fp32_baselines/preact_resnet18_imagenet_base_fp32.yaml) which you mention specifically: please notice that these are AlexNet with Batch Normalization and Preact ResNet18, respectively.  There are also TorchVision's [AlexNet](https://github.com/pytorch/vision/blob/master/torchvision/models/alexnet.py) and [ResNet18](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py) which are pretrained by Facebook (and unfortunately also don't have their training hyper-params published).  

@guyjacob - can you check if you have the hyper-params for [these](https://github.com/NervanaSystems/distiller/tree/master/examples/quantization/fp32_baselines)?

Thanks
Neta

**Comment by guyjacob** [comment]
Sorry for the delayed reply.
For pre-activation ResNet-18 the hyper-params we used were:
* Initial learning rate: 0.1
* Weight decay: 1e-4
* Momentum: 0.9
* Batch size: 256
* Num epochs: 110

(Note that other than the number of epochs, all of these values are the default values used by `compress_classifier.py`)

The command line to use (with everything specified explicitly):
`python compress_classifier.py --arch preact_resnet18 <path_to_imagenet_dataset> -p 50 --epochs 110 --compress=../quantization/fp32_baselines/preact_resnet18_imagenet_base_fp32.yaml -j 22 --vs 0 --lr 0.1 --wd 0.0001 --momentum 0.9 -b 256`

After 110 epochs we got 70.51% top-1 accuracy.

For AlexNet-BatchNorm, I found what I used, but something is not clear to me there, so I'll re-run to verify. When it's done I'll update here and push updated YAMLs.

Also, I'd like to emphasize @nzmora's comment that these 2 models are **not** the "vanilla" models that can be found in TorchVision. You can get those pre-trained by adding the `--pretrained` flag when running `compress_classifier.py`.  If you're wondering about the hyper-params used to train those, there's some info in their [readme](https://github.com/pytorch/examples/tree/master/imagenet#training) and in the their sample script [itself](https://github.com/pytorch/examples/blob/master/imagenet/main.py) (on which our `compress_classifier.py` is based).
